### ISSUE TITLE
OR operator

### ISSUE THREAD

**Issue by desile** [Infinidat/infi.clickhouse_orm#60]
Hello! 
Thanks for library!

I try to implement support of complex filters, but  the current way of filtration is not very suitable for this.
Do you plan to change it?

I made hotfix for myself, that acting like Q class, but self._fovs has hierarchical structure and init method expect hierarchical structure like this for example:

                {
                    'operation': 'AND',
                    'operands': [
                        {
                            'field': 'field1',
                            'type': 'in',
                            'value': ['a','b', 'c']
                        },
                        {
                            'field': 'field2',
                            'type': 'eq',
                            'value': 3
                        },
                        {
                            'operation': 'OR',
                            'operands': [
                                {
                                   'field': 'field3',
                                   'type': 'eq',
                                   'value': 'asdf'
                                },
                                {
                                    'field': 'field4',
                                    'type': 'lt',
                                    'value': 0
                                }
                           ]
                        }
                    ]
                }

I could not think of another solution that would fit in.
If you have any ideas of proper solution, but have not time for implementing it - I could help.

**Comment by ishirav** [comment]
Hi Paul,

Adding support for ORing between conditions is one of my TODO items... My plan was to make it look like Django's Q objects, where you can do things like:
```
conditions = Q(first_name='bob', last_name='smith') | Q(name='mary', last_name='jones')
Person.objects.filter(conditions)
```
I think this should be possible by enhancing our Q class:
- `__init__`: add self._mode which can be 'AND' or 'OR'
- implement `__or__` and `__and__` methods, which provide the "|" and "&" operators between Q objects. So q1|q2 should return a new Q object joining the conditions from q1 and q2 with an OR operator. That's the trickiest part, I guess.
- `to_sql`: when the Q object is made up of child Q objects, it should join their SQL using self._mode - something like `"({}) {} ({})".format(q1.to_sql(model_cls), self._mode, q2.to_sql(model_cls)`.

So if you want to take a crack at this and create a pull request, you're most welcome to do so. If not, I'll try to get to it soon.

**Comment by desile** [comment]
Nice way! Done it.
But for maintaining backward compatibility I have made named arg 'q' for Q object. It looks like:

clause = (Q(field1__between=[0, 2]) & Q(field2__eq='something')) | ~Q(id__gte=10)
result = TestTable.objects_in(clickhouse_db).only(TestTable.id.alias).filter(q=clause)

If it is ok I could make pull request in a short time.

**Comment by ishirav** [comment]
Why does it break backwards compatibility if you allow filter(clause) instead of filter(q=clause)? 
Actually filter(q=clause) breaks compatibility if you have a fields called "q" in your model...

**Comment by emakarov** [comment]
it's interesting whether django's code of Q object can be taken from django and adopted to infi.clickhouse_orm

**Comment by desile** [comment]
I cant call filter(clause) because of definition filter(self, **kwargs).
Or there are should be different method for that?

**Comment by ishirav** [comment]
I don't think anything will break if you change it to filter(self, *args, **kwargs)

**Comment by ishirav** [comment]
Released in v1.0.0
